src/avro/src/schema.rs

```diff
@@ -1018,20 +1018,21 @@ impl SchemaParser {
     fn parse_bytes(complex: &Map<String, Value>) -> Result<SchemaPiece, Error> {
         let logical_type = complex.get("logicalType").and_then(|v| v.as_str());
 
-        Ok(match logical_type {
-            Some("decimal") => {
-                let (precision, scale) = Self::parse_decimal(complex)?;
-                SchemaPiece::Decimal {
-                    precision,
-                    scale,
-                    fixed_size: None,
+        if let Some("decimal") = logical_type {
+            match Self::parse_decimal(complex) {
+                Ok((precision, scale)) => {
+                    return Ok(SchemaPiece::Decimal {
+                        precision,
+                        scale,
+                        fixed_size: None,
+                    })
                 }
+                Err(e) => debug!("{}", e),
```

Comment:
I think this should at least be `warn!` or `error!`

Reply:
agreed! fixing